### PR TITLE
Feature/change logic update view

### PR DESCRIPTION
**Pull request by 01Student10**
i   changed the  _updateSubject() method of the app_state.dart class, so the user can choose to which widget the method will be linked.
The original problem was that andi didnt want to switch to subject overview when the stop button was pressed. This happened because the stopp button called the Update Subject method.

### CODE REVIEW

File: lib/store/app_state.dart [hunk 1]
```diff
@@ -25,10 +25,11 @@ class AddNewSubject {
   final Subject subject;
   AddNewSubject(this.subject);
 }
-
+//TODO: add final Widget widget to this class
```
Comment (stigito) [inline lib/store/app_state.dart]: as far as i can see, the TODO is already implemented, right?

File: lib/store/app_state.dart [hunk 2]
```diff
@@ -47,9 +48,9 @@ class ChangeView {
 AppState _addNewSubject(AppState state, AddNewSubject action) =>
     new AppState(dbHelper: state.dbHelper, widget: new SubjectOverview())
       ..dbHelper.addNewSubject(action.subject);
-
+//TODO: widget: will get action.widget -that it will reviece from method call.
```
Comment (stigito) [inline lib/store/app_state.dart]: as far as i can see, the TODO is already implemented, right?

File: lib/presenter/subject_edit_or_delete.dart [hunk 3]
```diff
@@ -4,6 +4,7 @@ import 'package:easy_study/model/exam_type.dart';
 import 'package:easy_study/model/priority.dart';
```
Comment (stigito) [inline lib/presenter/subject_edit_or_delete.dart:4]: Please format all files you modified with strg + alt + L.

That is also the reason why travis fails.

**Review by stigito — CHANGES_REQUESTED** [review]

File: lib/presenter/subject_edit_or_delete.dart (hunk 3 above)
Comment (Plateria) [reply lib/presenter/subject_edit_or_delete.dart:4]: u could also right click on the lib folder inside your ide and click "Reformat Code with dartfmt..."

**Review by Plateria — APPROVED** [review]
Pls write tests when you added code.

**Review by stigito — APPROVED** [review]

**Pull request merged**